test/kvao/ttl.suite.js

```diff
@@ -0,0 +1,104 @@
+'use strict';
+
+var should = require('should');
+var helpers = require('./_helpers');
+var Promise = require('bluebird');
+
+module.exports = function(dataSourceFactory, connectorCapabilities) {
+  describe('ttl', function() {
+    var CacheItem;
+    beforeEach(function unpackContext() {
+      CacheItem = helpers.givenCacheItem(dataSourceFactory);
+    });
+
+    context('existing key with expire before expiration time', function() {
+      it('returns ttl - Callback API', function(done) {
+        CacheItem.set('a-key', 'a-value', 10, function(err) {
+          if (err) return done(err);
+          setTimeout(function() {
+            CacheItem.ttl('a-key', function(err, ttl) {
+              if (err) return done(err);
+              ttl.should.be.a.Number().lessThan(10).and.greaterThanOrEqual(0);
+              done();
+            });
+          }, 1);
+        });
+      });
+
+      it('returns ttl - Promise API', function() {
+        return CacheItem.set('a-key', 'a-value', 10)
+          .delay(1)
+          .then(function() {
+            return CacheItem.ttl('a-key');
+          })
+          .then(function(ttl) {
+            ttl.should.be.a.Number().lessThan(10).and.greaterThanOrEqual(0);
+          });
+      });
+    });
+
+    context('existing key with expire after expiration time', function(done) {
+      it('returns -2 - Callback API', function(done) {
+        CacheItem.set('a-key', 'a-value', 10, function(err) {
+          if (err) return done(err);
+          setTimeout(function() {
+            CacheItem.ttl('a-key', function(err, ttl) {
+              if (err) return done(err);
+              ttl.should.equal(-2);
+              done();
+            });
+          }, 11);
+        });
+      });
+
+      it('returns -2 - Promise API', function() {
+        return CacheItem.set('a-key', 'a-value', 10)
+          .delay(11)
```

Comment:
Please use a longer delay to prevent situation when slow machines and/or slow connections and/or other timing issues may cause the key to expire little bit later. I.e. use `20` here.